src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/ChainPal.cs

```diff
@@ -611,6 +611,12 @@ public static bool ReleaseSafeX509ChainHandle(IntPtr handle)
 
             SecTrustChainPal chainPal = new SecTrustChainPal();
 
+            // The allowNetwork controls all network activity for macOS chain building.
+            // There is no way to independently enable or disable online revocation checking
+            // and AIA fetching. If the caller specifies they want Online revocation checking,
+            // then we need to allow network operations (including AIA fetching.)
+            bool revocationRequiresNetwork = revocationMode == X509RevocationMode.Online;
```

Comment:
I seem to recall macOS does not support `Offline`. Does `Offline` act as `NoCheck` or `Online`? In other words, should this be:
```suggestion
            bool revocationRequiresNetwork = revocationMode != X509RevocationMode.NoCheck;
```

Reply:
That's a question for the ages :smile:.  I don't know what macOS does with CRL or OCSP caching, so it's hard to know which one is right.  Maybe something like

* With everything on do a revocation check
* Make sure the OCSP responses include the nextUpdate time (or whatever it's called) and it's at least 5 minutes in the future.
* Put the network state in no downloads (let Offline + DisableDownloads be "no downloads")
* Build the chain again

If that succeeds, then there's caching and we want Online vs not.  If that fails then we probably decide that we want NoCheck vs not, since we've let Offline be Online on macOS historically.

Reply:
Okay, given this line here, we opt in to the revocation checking polices as long as `!= NoCheck`. It's done that for a while, prior to .NET 5, so `Offline` was being treated as `Online`. 

https://github.com/dotnet/runtime/blob/cd7b2be873e81ca4fd191900f34ae95e9ab53476/src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/ChainPal.cs#L44

Given that, it probably makes sense to keep that behavior.

Reply:
> Given that, it probably makes sense to keep that behavior.

Well, previously we had two booleans that seemed indepdent:

* Do revocation checking
* Networky stuff (interpreted as "use AIA")

If validity-period OCSP means that there's a meaningful "revocation yes, download no" then Offline could be Offline if DisableCertificateDownloads is true, or Online when it's false.

But that's definitely a hard thing to test, document, and maintain, so I'm probably OK with "on macOS Offline means Online".